### ISSUE TITLE
Key of Number 0 Leads to Unexpected Result

### ISSUE THREAD

**Issue by philippfromme** [bpmn-io/min-dash#17]
__Describe the Bug__

Using keys of the type `number` can lead to unexpected results due to this check: https://github.com/bpmn-io/min-dash/blob/v3.6.0/lib/object.js#L44

__Steps to Reproduce__

```javascript
import { set } from "min-dash";

const target = {
  todos: [
    {
      name: "Buy milk"
    },
    {
      name: "Call mum"
    }
  ]
};

const path = ["todos", 0, "name"];

const value = "Buy eggs";

console.log(set(target, path, value)); // {0: Object, todos: "Buy eggs"}
```

CodeSandbox: https://codesandbox.io/s/min-dash-set-bug-18yhh?file=/src/index.js:0-276

__Expected Behavior__

`set` accepts keys of the type `number` as documented.

__Environment__

- Library version: 3.6.0

**Comment by nikku** [comment]
Gets fixed via #18, cf. https://github.com/bpmn-io/min-dash/pull/18/commits/682e3af6c2da525723c0d16c53b67720b2dcf8b7#diff-b95070f3e064df7b1b17f118735b2c1d3cc2112a81eaa4d7d66a33a80f89a128R331.

**Comment by philippfromme** [comment]
👏

**Comment by nikku** [comment]
Released fix via `v3.6.1`.

**Comment by philippfromme** [comment]
🚀
